Libraries/oneMKL/student_t_test/t_test_usm.cpp

```diff
@@ -0,0 +1,201 @@
+    res = 1;
+  } else {
+    res = 0;
+  }
+  // Free allocated memory
+  sycl::free(mean, queue);
+  sycl::free(variance, queue);
+  return res;
+}
+
+// T-test function with two input arrays
+// Returns: -1 if something went wrong, 1 - in case of NULL hypothesis should be
+// accepted, 0 - in case of NULL hypothesis should be rejected
+template <typename RealType>
+std::int32_t t_test(sycl::queue& queue, RealType* r1, std::int64_t n1,
+                    RealType* r2, std::int64_t n2) {
+  std::int32_t res = -1;
+  // Allocate memory to be passed inside oneMKL stats functions
+  RealType* mean1 = sycl::malloc_shared<RealType>(1, queue);
+  RealType* variance1 = sycl::malloc_shared<RealType>(1, queue);
+  RealType* mean2 = sycl::malloc_shared<RealType>(1, queue);
+  RealType* variance2 = sycl::malloc_shared<RealType>(1, queue);
+  // Perform computations of mean and variance
+  auto dataset1 =
+      oneapi::mkl::stats::make_dataset<oneapi::mkl::stats::layout::row_major>(
+          1, n1, r1);
+  auto dataset2 =
+      oneapi::mkl::stats::make_dataset<oneapi::mkl::stats::layout::row_major>(
+          1, n2, r2);
+  oneapi::mkl::stats::mean(queue, dataset1, mean1);
+  queue.wait_and_throw();
+  oneapi::mkl::stats::central_moment(queue, mean1, dataset1, variance1);
+  oneapi::mkl::stats::mean(queue, dataset2, mean2);
+  queue.wait_and_throw();
+  oneapi::mkl::stats::central_moment(queue, mean2, dataset2, variance2);
+  queue.wait_and_throw();
+  // Check the condition
+  bool almost_equal =
+      (variance1[0] < 2 * variance2[0]) || (variance2[0] < 2 * variance1[0]);
+  if (almost_equal) {
+    if ((sycl::abs(mean1[0] - mean2[0]) /
+         sycl::sqrt((static_cast<RealType>(1.0) / static_cast<RealType>(n1) +
+                     static_cast<RealType>(1.0) / static_cast<RealType>(n2)) *
+                    ((n1 - 1) * (n1 - 1) * variance1[0] +
+                     (n2 - 1) * (n2 - 1) * variance2[0]) /
+                    (n1 + n2 - 2))) < static_cast<RealType>(threshold)) {
+      res = 1;
+    } else {
+      res = 0;
+    }
+  } else {
+    if ((sycl::abs(mean1[0] - mean2[0]) /
+         sycl::sqrt((variance1[0] + variance2[0]))) <
+        static_cast<RealType>(threshold)) {
+      res = 1;
+    } else {
+      res = 0;
+    }
+  }
+  // Free allocated memory
+  sycl::free(mean1, queue);
+  sycl::free(variance1, queue);
+  sycl::free(mean2, queue);
+  sycl::free(variance2, queue);
+  return res;
+}
+
+int main(int argc, char** argv) {
+  std::cout << "\nStudent's T-test Simulation\n";
+  std::cout << "Unified Shared Memory Api\n";
+  std::cout << "-------------------------------------\n";
+
+  size_t n_points = n_samples;
+  fp_type mean = expected_mean;
+  fp_type std_dev = expected_std_dev;
+
+  if (argc >= 2) {
+    n_points = std::atol(argv[1]);
+    if (n_points == 0) {
+      n_points = n_samples;
+    }
+  }
+
+  if (argc >= 3) {
+    mean = std::atof(argv[2]);

```

Comment:
Error check on mean for bad command line?

Reply:
check was added